addons/udes_stock/models/stock_picking.py

```diff
@@ -1227,6 +1239,36 @@ def _new_picking_for_group(self, group_key, moves, move_lines):
 
         return picking
 
+    def apply_drop_location_policy(self):
+        """Apply suggested locations to move lines
+
+           raise ValidationError: if the policy set does not have
+                                  _allow_preprocess set
+        """
+
+        def check_policy_is_valid(policy):
```

Comment:
As you're not bringing anything from the outer method scope into a closure, I think it would be better to make this a free function instead of a nested one, so you won't be instantiating a new function object at each method invocation.